### DISCUSSION TITLE
NextAuth 4 and MongoDB 6

### DISCUSSION THREAD

**Discussion by fsa317** [nextauthjs/next-auth#10172] [Help]
I am trying to use NextAuth 4 with MongoDB 6.  My adapter is defined with     "@next-auth/mongodb-adapter": "^1.1.3",

I can get around the dependency issues but if I upgrade mongodb from 5.8 to ^6, magic links (email provider) stops working.

Is NextAuth 4 (using email provider) not compatible with MongoDB 6 (driver)?

**Comment by YuriGor** [comment]
Hi @fsa317 how have you got around dependency issue? 
I am trying to use next-auth with mongodb client v6.4 but npm complains latest 1.1.3 adapter requires 5 or 4 version.
I see [there is v2.5.0](https://github.com/nextauthjs/next-auth/releases/tag/%40auth%2Fmongodb-adapter%402.5.0) already but not published to npm yet?

**Reply by ananuness** [reply]
Definitely is not solved yet! Have you two thought about using Prisma? It can help a lot with those problems and prisma adapter is ok to use

**Reply by jynzo94** [reply]
The problem with email provider and mongodb version 6 is still not fixed. I lost a lot of time until codex found the bug... 

Currently this is the error I had:

```
https://next-auth.js.org/errors#adapter_error_useverificationtoken Cannot destructure property 'value' of '(intermediate value)' as it is null. { message: "Cannot destructure property 'value' of '(intermediate value)' as it is null.", stack: "TypeError: Cannot destructure property 'value' of '(intermediate value)' as it is null.\n" + ' at useVerificationToken (/app/.next/server/chunks/_f749d114._.js:1:2448)\n' + ' at process.processTicksAndRejections (node:internal/process/task_queues:95:5)', name: 'TypeError' } [next-auth][error][CALLBACK_EMAIL_ERROR] https://next-auth.js.org/errors#callback_email_error Cannot destructure property 'value' of '(intermediate value)' as it is null. Error [UseVerificationTokenError]: Cannot destructure property 'value' of '(intermediate value)' as it is null. at useVerificationToken (.next/server/chunks/_f749d114._.js:1:2448) { code: undefined }
```

Hope it will be indexed if someone search for
